File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def createHashMap(self, s: str) -> dict:
        res = defaultdict(int)
        for char in s:
            res[char] += 1
        return res

    def isAnagram(self, s: str, t: str) -> bool:
        str_s = self.createHashMap(s)
        str_t = self.createHashMap(t)
        return str_s == str_t
        #WCRT: O(N) | Space: O(N)

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string res = "";
        for(string str : strs){
            res += str + ".";
        }
        return res;
    }

    vector<string> decode(string s) {
        vector<string> res;
        string curr = "";
        for(char c : s){
            if(c == '.'){
                res.push_back(curr);
                curr = "";
            }else{
                curr += c;
            }
        }
        return res;
    }
};
